packages/ubuntu_test/lib/src/integration_test.dart

```diff
@@ -124,10 +124,24 @@ Future<bool> waitForWindowClosed() {
         break;
     }
   });
-
   return completer.future;
 }
 
+/// Sets an empty mock method call handler for window close requests.
+/// onDestroy is called upon response of the method call handler.
+void mockOnceWindowClose({void Function()? onDone}) {
```

Comment:
so what was the underlying issue? that multiple instances were created?

Reply:
Making the ApplyingChangesPage being the last one means it will always close the window. WIthout mocking the method call handler, it closes the window sometime later. If, instead of the empty mock, I just `await testApplyingChangesPage(tester, expectClose: true)` (which contains a mock) the test hangs forever.

Reply:
Sorry for being so thick in the head but... why does it hang? :thinking:

Reply:
The post frame callback! Now that we are not triggering rebuilds we no longer need that. Thank you for the insights!